src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,40 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given address.
+     *
+     * @throws IllegalValueException if given address string is invalid.
+     */
+    public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        this.value = contact;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
this would cause:

    Contact address = new Address(...);
    Contact email = new Email(...);
    email.equals(address); // returns true.